pkgs/applications/office/libreoffice/wrapper.nix

```diff
@@ -1,19 +1,33 @@
-{ libreoffice, runCommand, dbus, bash }:
-let
-  jdk = libreoffice.jdk;
-in
-(runCommand libreoffice.name {
-  inherit dbus libreoffice jdk bash;
-} ''
-  mkdir -p "$out/bin"
-  ln -s "${libreoffice}/share" "$out/share"
-  substituteAll "${./wrapper.sh}" "$out/bin/soffice"
-  chmod a+x "$out/bin/soffice"
+{ lib, stdenvNoCC
+, libreoffice, dbus, bash, substituteAll
+, dolphinTemplates ? false
+}:
+stdenvNoCC.mkDerivation {
+  inherit (libreoffice) name src jdk meta;
 
-  for i in $(ls "${libreoffice}/bin/"); do
-    test "$i" = "soffice" || ln -s soffice "$out/bin/$(basename "$i")"
-  done
-'') // {
-  inherit libreoffice dbus;
-  meta = libreoffice.meta;
+  inherit dbus libreoffice bash;
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  installPhase = ''
```

Comment:
When overriding `installPhase`, you should also provide `runHook preInstall` and `runHook postInstall`. Though, in this case, it is better to rely on the default `installPhase`. See <https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md> for more info.

Reply:
Thanks. This is a wrapper previously used `runCommand`. However, I thought I had to change it to `mkDerivation` to get access to the unpackPhase to extract the necessary file. But thanks for the link, I didnt know that I could manually unpack the file using `unpack`.

Going to switch it back to runCommand